src/event_message/mod.rs

```diff
@@ -78,7 +75,7 @@ impl EventMessage {
         let icp_event_data = Event {
             prefix: IdentifierPrefix::SelfAddressing(code.derive(&[0u8; 32])),
             sn: 0,
-            event_data: EventData::Icp(icp.clone()),
+            event_data: inceptive_event_data.clone(),
```

Comment:
to prevent passing in `Rot` or other event data types, we should either check the enum here or define two separate fns like `get_inception_data(&InceptionEvent...` and `get_delegated_inception_data(&DelegatedInceptionEvent...`. if we define two separate ones we can also avoid the `.clone()` in the `incept_self_addressing` functions